test/nightly_test.go

```diff
@@ -0,0 +1,209 @@
+}
+
+func prometheusMetricsFullURL(localport int) string {
+	queryURL := "/api/v1/query_range?"
+	queryParams := strings.Join([]string{
+		"query=up",
+		// take a 5 min range to prevent flakes because data points are scattered over time
+		"start=" + time.Now().Add(-5*time.Minute).Format(time.RFC3339),
+		"end=" + time.Now().Format(time.RFC3339),
+		"step=1s",
+	}, "&")
+	return fmt.Sprintf("http://localhost:%d%s", localport, queryURL+queryParams)
+}
+
+type prometheusUpMetricsResponse struct {
+	Status string         `json:"status"`
+	Data   prometheusData `json:"data"`
+}
+
+type prometheusData struct {
+	ResultType string                        `json:"resultType"`
+	Result     []prometheusUpMetricWithValue `json:"result"`
+}
+
+type prometheusUpMetricWithValue struct {
+	Metric PrometheusUpMetric `json:"metric"`
+}
+
+type PrometheusUpMetric struct {
+	Name      string `json:"__name__"`
+	App       string `json:"app"`
+	Namespace string `json:"namespace"`
+	Service   string `json:"service"`
+	Job       string `json:"job"`
+}
+
+func (p *PrometheusUpMetric) key() string {
+	return strings.Join([]string{p.Name, p.App, p.Namespace, p.Service, p.Job}, " ; ")
+}
+
+func unmarshallPrometheusMetrics(data []byte) ([]PrometheusUpMetric, error) {
+	resp := &prometheusUpMetricsResponse{}
+	if err := json.Unmarshal(data, resp); err != nil {
+		return nil, fmt.Errorf("error unmarshallPrometheusMetrics: %w", err)
+	}
+
+	if resp.Data.ResultType != "matrix" {
+		return nil, fmt.Errorf("resultType of data is not matrix. got: %v", resp.Data.ResultType)
+	}
+	return summarizePrometheusMetrics(resp.Data.Result), nil
+}
+
+func summarizePrometheusMetrics(metricsWithValue []prometheusUpMetricWithValue) []PrometheusUpMetric {
+	keyToMetric := map[string]PrometheusUpMetric{}
+
+	for _, metricWithValue := range metricsWithValue {
+		metric := metricWithValue.Metric
+		keyToMetric[metric.key()] = metric
+	}
+
+	metrics := make([]PrometheusUpMetric, len(keyToMetric))
+	i := 0
+	for _, metric := range keyToMetric {
+		metrics[i] = metric
+		i++
+	}
+	sort.Slice(metrics, func(i, j int) bool {
+		return metrics[i].key() < metrics[j].key()
+	})
+
+	return metrics
+}
+
+func updatePrometheusMetricFixtures(filename string, expectedPromMetrics []PrometheusUpMetric) error {
+	b, err := yaml.Marshal(expectedPromMetrics)
+	if err != nil {
+		return err
+	}
+	fmt.Println(filename + " is now updated with:")
+	fmt.Println(string(b))
+	return ioutil.WriteFile(filename, b, 0600)
+}
+
+func testMetricFixtures(filename string, gotPromMetricsTests []PrometheusUpMetric) error {
+	b, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	var expectedPromMetricsTests []PrometheusUpMetric
+	err = yaml.Unmarshal(b, &expectedPromMetricsTests)
+	if err != nil {
+		return err
+	}
+
+	if !reflect.DeepEqual(gotPromMetricsTests, expectedPromMetricsTests) {
+		return fmt.Errorf(cmp.Diff(gotPromMetricsTests, expectedPromMetricsTests))
+	}
+	return nil
+}
```

Comment:
you might consider renaming the file `test/nightly_prometheus_test.go` just to make it easier to hone in on.

Reply:
good idea. done.